dipy/reconst/dki_micro.py

```diff
@@ -0,0 +1,647 @@
+
+        References
+        ----------
+        .. [1] Fieremans, E., Jensen, J.H., Helpern, J.A., 2011. White Matter
+               Characterization with Diffusion Kurtosis Imaging. Neuroimage
+               58(1): 177-188. doi:10.1016/j.neuroimage.2011.06.006
+        """
+        DiffusionKurtosisFit.__init__(self, model, model_params)
+
+    @property
+    def awf(self):
+        """ Returns the volume fraction of the restricted diffusion compartment
+        also known as axonal water fraction.
+
+        Note
+        ----
+        The volume fraction of the restricted diffusion compartment can be seem
+        as the volume fraction of the intra-cellular compartment [1]_.
+
+        References
+        ----------
+        .. [1] Fieremans, E., Jensen, J.H., Helpern, J.A., 2011. White Matter
+               Characterization with Diffusion Kurtosis Imaging. Neuroimage
+               58(1): 177-188. doi:10.1016/j.neuroimage.2011.06.006
+        """
+        return self.model_params[..., 27]
+
+    @property
+    def restricted_evals(self):
+        """ Returns the eigenvalues of the restricted diffusion compartment.
+
+        Note
+        -----
+        The restricted diffusion tensor can be seem as the tissue's
+        intra-cellular diffusion compartment [1]_.
+
+        References
+        ----------
+        .. [1] Fieremans, E., Jensen, J.H., Helpern, J.A., 2011. White Matter
+               Characterization with Diffusion Kurtosis Imaging. Neuroimage
+               58(1): 177-188. doi:10.1016/j.neuroimage.2011.06.006
+        """
+        self._is_awfonly()
+        return _compartments_eigenvalues(self.model_params[..., 34:40],
+                                         self._min_diffusivity)
+
+    @property
+    def hindered_evals(self):
+        """ Returns the eigenvalues of the hindered diffusion compartment.
+
+        Note
+        -----
+        The hindered diffusion tensor can be seem as the tissue's
+        extra-cellular diffusion compartment [1]_.
+
+        References
+        ----------
+        .. [1] Fieremans, E., Jensen, J.H., Helpern, J.A., 2011. White Matter
+               Characterization with Diffusion Kurtosis Imaging. Neuroimage
+               58(1): 177-188. doi:10.1016/j.neuroimage.2011.06.006
+        """
+        self._is_awfonly()
+        return _compartments_eigenvalues(self.model_params[..., 28:34],
+                                         self._min_diffusivity)
+
+    @property
+    def axonal_diffusivity(self):
+        """ Returns the axonal diffusivity defined as the restricted diffusion
+        tensor trace [1]_.
+
+        References
+        ----------
+        .. [1] Fieremans, E., Jensen, J.H., Helpern, J.A., 2011. White Matter
+               Characterization with Diffusion Kurtosis Imaging. Neuroimage
+               58(1): 177-188. doi:10.1016/j.neuroimage.2011.06.006
+        """
+        return trace(self.restricted_evals)
```

Comment:
No need! This is calling self.restricted_evals which checks _is_awf_only for it!